### DISCUSSION TITLE
Does Ops or nanos produce a single, executable binary file that can be run independently of the ops tool itself?

### DISCUSSION THREAD

**Discussion by armenr** [nanovms/ops#1599] [General]
^^ As stated in the question title: Does Ops or nanos produce a single, executable binary file that can be run independently of the ops tool itself?

**Comment by eyberg** [comment]
ops exports a disk image that can be ran on any cloud/hypervisor - it doesn't export binary elfs; when deploying to the cloud ops doesn't actually 'run it' either - it talks to the various cloud/hypervisor apis so technically once you have the image you can just go through the motions yourself but we don't encourage that since there are lots of little checks to ensure things like proper instance type, naming, etc.

**Reply by armenr** [reply]
Thank you!
